internal/verifier/web_server.go

```diff
@@ -177,15 +185,38 @@ func (server *WebServer) Run(ctx context.Context) error {
 // EmptyRequest is for request with empty body
 type EmptyRequest struct{}
 
+// CheckRequest is for requests to the /check endpoint.
+type CheckRequest struct {
+	Filter map[string]any `json:"filter"`
+}
+
 func (server *WebServer) checkEndPoint(c *gin.Context) {
-	var json EmptyRequest
+	var req CheckRequest
 
-	if err := c.ShouldBindJSON(&json); err != nil {
+	if err := c.ShouldBindJSON(&req); err != nil {
+		err = errors.Wrap(err, "filter is not valid JSON")
 		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
 		return
 	}
 
-	server.Mapi.Check(context.Background())
+	var filter bson.D
+	if req.Filter != nil {
+		raw, err := bson.Marshal(req.Filter)
+		if err != nil {
+			err = errors.Wrap(err, "failed to marshal json to bson.Raw")
```

Comment:
Nit: This isn’t marshalling _JSON_ per se. The fact that the `map[string]any` came from JSON isn’t relevant here.
```suggestion
			err = errors.Wrap(err, "failed to marshal map to bson.Raw")
```

Reply:
No longer relevant because we don't marshal map here.